crosschain-reverse-resolver/contracts/DefaultReverseResolver.sol

```diff
@@ -0,0 +1,95 @@
+pragma solidity >=0.8.4;
+
+import "./IDefaultReverseResolver.sol";
+import "@ensdomains/ens-contracts/contracts/reverseRegistrar/SignatureReverseResolver.sol";
+import "@ensdomains/ens-contracts/contracts/wrapper/BytesUtils.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import {ITextResolver} from "@ensdomains/ens-contracts/contracts/resolvers/profiles/ITextResolver.sol";
+import {INameResolver} from "@ensdomains/ens-contracts/contracts/resolvers/profiles/INameResolver.sol";
+import "@ensdomains/ens-contracts/contracts/utils/HexUtils.sol";
+import "@ensdomains/ens-contracts/contracts/resolvers/profiles/IExtendedResolver.sol";
+
+/**
+ * A fallback reverser resolver to resolve when L2 reverse resolver has no names set.
+ * The contract will be set under "default.reverse" namespace
+ * It can only be set by EOA as contract accounts are chain dependent.
+ */
+contract DefaultReverseResolver is
+    Ownable,
+    IDefaultReverseResolver,
+    IExtendedResolver,
+    ERC165,
+    SignatureReverseResolver
+{
+    uint256 constant ADDRESS_LENGTH = 40;
+    using ECDSA for bytes32;
+    using BytesUtils for bytes;
+    // The namehash of 'default.reverse'
+    bytes32 private constant DEFAULT_REVERSE_NODE =
+        0x53a2e7cce84726721578c676b4798972d354dd7c62c832415371716693edd312;
+
+    /**
+     * @dev Constructor
+     */
+    constructor() SignatureReverseResolver(DEFAULT_REVERSE_NODE, 0) {}
```

Comment:
In this case, I am using `coinType = 0` to represent a default name space assuming no one is going to use Bitcoin cointype 0 for l2 reverse resolver